src/app/base/components/dropdown/dropdown.component.ts

```diff
@@ -218,6 +218,9 @@ export class DropdownComponent implements OnChanges {
         c.get('active').value &&
         (!hasNegate || (hasNegate && !c.get('negate').value))
       ) {
+        if (this.dropdownOptions.hasSelectAllOption) {
+          if (c.get('id').value == 'SELECT_ALL') continue;
+        }
```

Comment:
The logic in this for loop seems to be quite complex for me with many if-elses and nested cases. I would think of maybe refactoring to use the same structure but outsource the checks into expressive functions so it is easy understandable what happens in this loop :) This is like always my opinion, but to give you an idea what my perspective is:

I am not that familiar with our frontend development at all but what I see is something like this (Attention: Javascript like peusdocode :D) :

```
if (labelFor == 'EMPTY' && c.get('active').value) ...
else if (
  labelFor == 'NOT_NEGATED' &&
  c.get('active').value &&
  (!hasNegate || (hasNegate && !c.get('negate').value))
) ... 
else if (
  labelFor == 'NEGATED' &&
  c.get('active').value &&
  hasNegate &&
  c.get('negate').value
) ... 
  if (
    !atLeastOneNegated &&
    c.get('active').value &&
    hasNegate &&
    c.get('negate').value
  ) ....
```

this could be rewritten into something like:
```
if(isActiveControlEmpty(x,y)) ...
elseif(isActiveControlNotNegate(x,y)) ...
elseif(isActiveControlNegated(x,y)) ...
else
```

I hope this is understandable :D So for me it would be way easier to understand what this method and the steps should do if it is written in a more (simon-) readable way, but like I said, just a perspective which can or can not be relevant for you :)

Reply:
I would keep it like this for now, maybe add some comments because I think we would do a lot of passing of the parameters for only checking the condition. Not sure if that's a better approach tho :)